### PR TITLE
Fix catalyst dependency on system git.

### PR DESCRIPTION
**Pull request by sidml**
## Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contribution guide](https://github.com/catalyst-team/catalyst/blob/master/CONTRIBUTING.md)?
- [x] Did you check the code style? `catalyst-make-codestyle && catalyst-check-codestyle` (`pip install -U catalyst-codestyle`).
- [ ] Did you make sure to update the docs? We use Google format for all the methods and classes.
- [ ] Did you check the docs with `make check-docs`?
- [ ] Did you write any new necessary tests?
- [ ] Did you add your new functionality to the docs?
- [ ] Did you update the [CHANGELOG](https://github.com/catalyst-team/catalyst/CHANGELOG.md)?
- [x] **You can use 'Login as guest' to see Teamcity build logs.**

## Related Issue
Fix related to issue [847](https://github.com/catalyst-team/catalyst/issues/847).

## Type of Change

- [ ] Examples / docs / tutorials / contributors update
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves an existing feature)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

### CODE REVIEW

File: catalyst/dl/utils/wizard.py [hunk 1]
```diff
@@ -5,8 +5,10 @@
 import yaml
 
 from catalyst import registry
-from catalyst.utils import clone_pipeline, import_module
-from catalyst.utils.pipelines import URLS
+from catalyst.tools.settings import IS_GIT_AVAILABLE
```
Comment (Scitator) [inline catalyst/dl/utils/wizard.py]: are you sure, that wizard is working correctly without `git`?

Comment (sidml) [reply catalyst/dl/utils/wizard.py]: Yes, seems to work correctly on my pc.

Comment (sidml) [reply catalyst/dl/utils/wizard.py]: Without the git check in wizard.py i get the following error

> python -c "from catalyst.dl import SupervisedRunner, State, CallbackOrder, Callback, CheckpointCallback"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/sid/.local/lib/python3.8/site-packages/catalyst/dl/__init__.py", line 5, in <module>
    from catalyst.dl.callbacks import *
  File "/home/sid/.local/lib/python3.8/site-packages/catalyst/dl/callbacks/__init__.py", line 6, in <module>
    from catalyst.dl.callbacks.meter import MeterMetricsCallback
  File "/home/sid/.local/lib/python3.8/site-packages/catalyst/dl/callbacks/meter.py", line 7, in <module>
    from catalyst.dl.utils import get_activation_fn
  File "/home/sid/.local/lib/python3.8/site-packages/catalyst/dl/utils/__init__.py", line 20, in <module>
    from catalyst.dl.utils.wizard import run_wizard, Wizard
  File "/home/sid/.local/lib/python3.8/site-packages/catalyst/dl/utils/wizard.py", line 8, in <module>
    from catalyst.utils import clone_pipeline, import_module
ImportError: cannot import name 'clone_pipeline' from 'catalyst.utils' (/home/sid/.local/lib/python3.8/site-packages/catalyst/utils/__init__.py)

You can replicate this error using the following commands in a conda env:
1. conda create -n git-test python=3.7
2. conda activate git-test
3. pip install git+https://github.com/sidml/catalyst.git
4. sudo apt remove git
5. python -c "from catalyst.dl import SupervisedRunner, State, CallbackOrder, Callback, CheckpointCallback"
The above code should run correctly.
If you remove the git checks from wizard.py, you should be able to replicate the above mentioned error.

Comment (Scitator) [reply catalyst/dl/utils/wizard.py]: the thing is that, without this imports the wizard would not work :)

could you please move git check to https://github.com/catalyst-team/catalyst/blob/c8f058f0792f549695227d2a6768b5ce05a4a9ba/catalyst/dl/utils/__init__.py#L20

and also to main here
https://github.com/catalyst-team/catalyst/blob/2f4936b116cd1db88e7739e62540878404f1be5a/catalyst/dl/__main__.py#L7

like here
https://github.com/catalyst-team/catalyst/blob/2f4936b116cd1db88e7739e62540878404f1be5a/catalyst/data/__main__.py#L87
but only with if logic :)

**Review by Scitator — DISMISSED** [review]

**Comment by Scitator** [comment]
btw, @sidml could you please also check the codestyle
https://github.com/catalyst-team/catalyst/blob/master/CONTRIBUTING.md#codestyle
there are some errors from the CI

File: catalyst/dl/utils/wizard.py (hunk 1 above)
Comment (sidml) [reply catalyst/dl/utils/wizard.py]: Thanks ! Will do.

Comment (sidml) [reply catalyst/dl/utils/wizard.py]: In https://github.com/catalyst-team/catalyst/blob/2f4936b116cd1db88e7739e62540878404f1be5a/catalyst/data/__main__.py#L87

Should i do something like this ?
```
import logging
logger = logging.getLogger(__name__)
from catalyst.tools.settings import IS_GIT_AVAILABLE
if IS_GIT_AVAILABLE:
    COMMANDS = OrderedDict([("init", init), ("run", run), ("trace", trace)])
else:
    logger.error("git is not available,"
            + " to install git, run `sudo apt install git`."
        )
    raise ImportError
```

Comment (Scitator) [reply catalyst/dl/utils/wizard.py]: a bit easier
```
COMMANDS = OrderedDict([("run", run), ("trace", trace)])

if IS_GIT_AVAILABLE:
  from .. import init
  COMMANDS['init'] = init
```

Comment (sidml) [reply catalyst/dl/utils/wizard.py]: What is the correct way to check codestyle ?
I tried this `catalyst-make-codestyle && catalyst-check-codestyle`and this is the output.
image
But in CI, it's still failing.. I wonder why.

Comment (Scitator) [reply catalyst/dl/utils/wizard.py]: are you using last version of codestyle?
moreover, you can also simply check the CI output, there are only a few issues with imports https://github.com/catalyst-team/catalyst/pull/922/checks?check_run_id=1021029262

**Comment by Scitator** [comment]
@sidml Thank you for your Pull Request!

**Comment by sidml** [comment]
@Scitator Thanks :-)

**Pull request merged**